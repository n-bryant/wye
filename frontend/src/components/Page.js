import React from "react";
import PropTypes from "prop-types";

import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { withStyles } from "@material-ui/core/styles";
import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../lib/classNamePrefix";
import theme from "../../lib/theme";

import Meta from "./Meta";
import Header from "./Header";

import styles from "./Page.styles";

/**
 * wraps page content with material-ui theme, css reset, and meta details
 */
export const Page = props => {
  const classnames = Page.classnames(props);

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline>
        <Meta />
        <div className={classnames.root()}>
          <Header />
          <div className={classnames.element("innerContent")}>
            {props.children}
          </div>
        </div>
      </CssBaseline>
    </ThemeProvider>
  );
};
Page.classnames = createClassNameHelper(`${JSS_CLASS_NAME_PREFIX}Page`);
Page.propTypes = {
  // styles applied to the Page
  classes: PropTypes.shape({
    root: PropTypes.string,
    innerContent: PropTypes.string
  })
};
Page.defaultProps = {
  classes: {}
};

export default withStyles(styles)(Page);
