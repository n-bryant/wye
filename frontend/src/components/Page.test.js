import { Page } from "./Page";
import toJson from "enzyme-to-json";
import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";

describe("Page", () => {
  it("should render its children wrapped with the material theme and css reset", () => {
    const props = {
      classes: {
        root: "root",
        innerContent: "innerContent"
      },
      children: () => <div>foo</div>
    };
    const wrapper = shallow(<Page {...props} />);
    expect(wrapper.find(ThemeProvider)).toHaveLength(1);
    expect(wrapper.find(CssBaseline)).toHaveLength(1);
    const content = toJson(
      wrapper.findWhere(n => n.hasClass(props.classes.root))
    );
    expect(content).toMatchSnapshot();
  });
});
