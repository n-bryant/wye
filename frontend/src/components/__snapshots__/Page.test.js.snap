// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render its children wrapped with the material theme and css reset 1`] = `
<div
  className="wye-Page root"
>
  <WithStyles(Header)
    classes={Object {}}
  />
  <div
    className="wye-Page__innerContent innerContent"
  >
    <Component />
  </div>
</div>
`;
